tests/integration/actions/test_generate_data_quality_check_expectations.py

```diff
@@ -32,64 +33,95 @@ def user_api_token_headers_org_admin_sc_org():
 
-    table_data_asset = data_source.add_table_asset(
-        table_name="checkpoints", name="local-mercury-db-checkpoints-table"
-    )
+    table_data_asset_name = "local-mercury-db-checkpoints-table"
+    try:
+        table_data_asset = data_source.get_asset(table_data_asset_name)
+    except LookupError:
+        table_data_asset = data_source.add_table_asset(
+            table_name="checkpoints", name=table_data_asset_name
+        )
 
-    suite = context.suites.add(ExpectationSuite(name="local-mercury-db-checkpoints-table Suite"))
+    suite_name = "local-mercury-db-checkpoints-table Suite"
+    try:
+        suite = context.suites.get(suite_name)
+    except DataContextError:
+        suite = context.suites.add(ExpectationSuite(name=suite_name))
 
     # Create validation
-    batch_definition = table_data_asset.add_batch_definition_whole_table(
-        name="local-mercury-db-checkpoints-table-batch-definition"
-    )
-    validation = context.validation_definitions.add(
-        ValidationDefinition(
-            name="local-mercury-db-checkpoints-table Validation",
-            suite=suite,
-            data=batch_definition,
+    batch_definition_name = "local-mercury-db-checkpoints-table-batch-definition"
+    try:
+        batch_definition = table_data_asset.get_batch_definition(batch_definition_name)
+    except KeyError:
+        batch_definition = table_data_asset.add_batch_definition_whole_table(
+            name=batch_definition_name
         )
-    )
 
-    # Mark the validation as gx_managed
-    engine = sa.create_engine("postgresql://postgres:postgres@localhost:5432/mercury")
-    with engine.begin() as conn:
-        query = f"UPDATE validations SET gx_managed=true WHERE id='{validation.id}'"
-        conn.execute(sa.text(query))
-        conn.commit()
+    validation_name = "local-mercury-db-checkpoints-table Validation"
+    try:
+        validation = context.validation_definitions.get(validation_name)
+    except DataContextError:
+        validation = context.validation_definitions.add(
+            ValidationDefinition(
+                name=validation_name,
+                suite=suite,
+                data=batch_definition,
+            )
+        )
 
-    # Mark the suite as gx_managed
-    with engine.begin() as conn:
-        query = f"UPDATE expectation_suites SET gx_managed=true WHERE id='{suite.id}'"
-        conn.execute(sa.text(query))
+    engine = sa.create_engine("postgresql://postgres:postgres@localhost:5432/mercury")
+    with engine.connect() as conn:
+        # Check if there's already a gx_managed=true validation for this asset
+        existing_gx_managed_validations = conn.execute(
+            sa.text(
+                f"SELECT COUNT(*) FROM validations WHERE asset_ref_id='{table_data_asset.id}' AND gx_managed=true"
+            )
+        ).scalar()
+
+        # Only set gx_managed=true if there isn't already one for this asset
+        if not existing_gx_managed_validations:
```

Comment:
This indicates that the environment is not deterministic. If there are conditionals in the test fixtures, I can't trust what the tests are doing